### PR TITLE
fix: helm release login not needed when using OCI + docker-credential-helpers.

### PR DESCRIPTION
**Pull request by babadofar**
I have tested this and it can deploy to AWS ECR registry.
It needs the docker-credential helpers available on the path and credentials  to push to OCR registry. Similar to how regular containers are pushed to registry. So some changes to how the image that runs this is needed.

### CODE REVIEW

**Comment by babadofar** [comment]
sonarcloud complains on duplicate code in tests, again...

File: pkg/cmd/helm/release/release_test.go [hunk 1]
```diff
@@ -303,7 +302,7 @@ func TestStepHelmReleaseWithOCINoOCILogin(t *testing.T) {
 		},
 
 		fakerunner.FakeResult{
-			CLI: "helm push myapp-" + chartVersion + ".tgz " + OCIRegistry,
+			CLI: "helm push myapp-" + chartVersion + ".tgz " + OCIRegistry + " --registry-config /tekton/creds-secrets/tekton-container-registry-auth/.dockerconfigjson",
```
Comment (tomhobson) [inline pkg/cmd/helm/release/release_test.go:305]: should we not use the o.RegistryConfigFile here?

File: pkg/cmd/helm/release/release_test.go [hunk 2]
```diff
@@ -230,7 +231,7 @@ func TestStepHelmReleaseWithOCIUsingUserName(t *testing.T) {
 			CLI: "helm registry login " + OCIRegistry + " --username " + o.RepositoryUsername + " --password " + o.RepositoryPassword,
 		},
 		fakerunner.FakeResult{
-			CLI: "helm push myapp-" + chartVersion + ".tgz " + OCIRegistry,
+			CLI: "helm push myapp-" + chartVersion + ".tgz " + OCIRegistry + " --registry-config /tekton/creds-secrets/tekton-container-registry-auth/.dockerconfigjson",
```
Comment (tomhobson) [inline pkg/cmd/helm/release/release_test.go:234]: Same here? o.RegistryConfigFile

**Review by tomhobson — CHANGES_REQUESTED** [review]

File: pkg/cmd/helm/release/release_test.go (hunk 2 above)
Comment (babadofar) [reply pkg/cmd/helm/release/release_test.go:234]: ok

Comment (babadofar) [reply pkg/cmd/helm/release/release_test.go:234]: quite smooth to get the comments in the file immediately. VS Code

**Comment by babadofar** [comment]
It works for me now, but the setting of config is a myriad of possibilities and quite hard to test all, and to reason about what is going to happen.

**Review by tomhobson — APPROVED** [review]

**Pull request merged**